sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/codegen/ContractDecoder.java

```diff
@@ -21,18 +22,40 @@ public ContractDecoder(
       Iterable<? extends ContractCompanion<? extends Contract<?, ?>, ?, ? extends DamlRecord<?>>>
           companions) {
     this.companions =
-        StreamSupport.stream(companions.spliterator(), false)
-            .collect(Collectors.toMap(c -> c.TEMPLATE_ID, Function.identity()));
+        Stream.concat(
+                // Each companion should be keyed by TEMPLATE_ID and by TEMPLATE_ID_WITH_PACKAGE_ID
+                StreamSupport.stream(companions.spliterator(), false)
+                    .map(c -> Map.entry(c.TEMPLATE_ID, c)),
+                StreamSupport.stream(companions.spliterator(), false)
+                    .map(c -> Map.entry(c.TEMPLATE_ID_WITH_PACKAGE_ID, c)))
```

Comment:
we can use a single stream and avoid the concat
``` 
   StreamSupport.stream(companions.spliterator(), false)
       .flatMap(c -> Stream.of(
            Map.entry(c.TEMPLATE_ID, c),
            Map.entry(c.TEMPLATE_ID_WITH_PACKAGE_ID, c)
       ))
```

Reply:
Done